actors/smarty-daily/main.js

```diff
@@ -0,0 +1,79 @@
+import { Actor } from "apify";
+import { LinkeDOMCrawler, createLinkeDOMRouter } from "@crawlee/linkedom";
+import rollbar from "@hlidac-shopu/actors-common/rollbar.js";
+import { uploadToKeboola } from "@hlidac-shopu/actors-common/keboola.js";
+import { cleanPriceText } from "@hlidac-shopu/lib/parse.mjs";
+
+// Not using Labels from actors-common cause I need two separate labels for listing
+const LABEL_CUSTOM = {
+  INDEX: `INDEX`,
+  LISTING_PREFLIGHT: `LISTING_PREFLIGHT`,
+  LISTING: `LISTING`,
+}
+
+const baseUrl = "https://www.smarty.cz";
+
+async function main() {
+  rollbar.init();
+  const crawler = new LinkeDOMCrawler({
+    requestHandler: createLinkeDOMRouter({
+      [LABEL_CUSTOM.INDEX]: async ({ window: { document }, crawler }) => {
+        const urls = Array.from(document.querySelectorAll(`url loc`)).map((el) => el.innerText);
+        // https://www.smarty.cz/Vyrobce/samsung - we wanna just these
+        // https://www.smarty.cz/Vyrobce/samsung/mobilni-telefony - but not these deep urls
+        const brandsUrls = urls.filter(url => url.split(`/`).length === 5);
+        const { addedRequests } = await crawler.addRequests(brandsUrls.map(url => ({
+          url: `${url}#us=1&s=n&pg=all`,
+          label: LABEL_CUSTOM.LISTING_PREFLIGHT
+        })));
+      },
+      [LABEL_CUSTOM.LISTING_PREFLIGHT]: async ({ window: { document }, crawler }) => {
+        const producerId = document.querySelector(`#fId`).value;
+        const producerAlias = document.querySelector(`#producerAlias`).value;
+        const url = `https://www.smarty.cz/Product/ProducerProductItems?${(new URLSearchParams({
+          producerId,
+          producerAlias,
+          itemTypeId: ``,
+          productFilter: `1`,
+          paramJson: JSON.stringify({ us: `1`, s: `n`, pg: `all` }),
+          sort: `n`,
+          page: `all`
+        }))}`;
+        await crawler.addRequests([{
+          url,
+          label: LABEL_CUSTOM.LISTING
+        }], {
+          forefront: true // makes debugging easier, cause it reveals problems in LISTING handler faster
+        });
+      },
+      [LABEL_CUSTOM.LISTING]: async ({ window: { document }, log }) => {
+        if (document.querySelector(`.errorPage`)) {
+          log.error(`errorPage`, { html: document.innerHTML });
+          return;
+        }
+        const products = [];
+        document.querySelectorAll(`.item`).forEach((el) => {
```

Comment:
V bloku index používáš `Array.from().map()` tady pro změnu `.forEach()` s mutacema pole. Prosím, udělej to konzistentně bez mutací.

Reply:
kvOK https://github.com/topmonks/hlidac-shopu/pull/1151/commits/5f9165ee753f8bd1b290704911ed1f9a22052842